src/prediction/_model_export_manager.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2016 TensorLab. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+
+import os
+import collections
+import time
+import tensorflow as tf
+from tensorflow.python.lib.io import file_io
+import tensorfx as tfx
+
+class ModelExportManager(object):
+  def __init__(self, intermediate_model_folder, final_model_folder, exports_to_keep):
+    """Exports last model into final location, keeps last few trained models.
```

Comment:
We are already writing out checkpoints along the way, and its actually straightforward to go from checkpoint to a model if needed (I wish a saved model used the checkpoint file as-is ... maybe there is a way), so this work of loading each checkpoint to save a saved model is quite wasteful (once you factor in larger models and reads/writes to GCS). So I am concerned with this change at a broader level.

Additionally the pattern of keeping N changes is problematic on GCS, since they require enumeration/copies. Something other than the training process should be in charge of deletes based on size/age/quality.

In fact what we need with multiple checkpoints/exports along the way is some place to store the eval metric achieved with that checkpoint, so a user can decide which checkpoint to use easily.

So I think what we really need is just an argument to expose the # of checkpoints to keep, and default it to 0, meaning keep-all, that is passed in into the saver construction.